src/helper/remote_config/protocol/tuf/parser.cpp

```diff
@@ -0,0 +1,392 @@
+parse(std::string &&body)
+{
+    rapidjson::Document serialized_doc;
+    std::pair<remote_config_parser_result, std::optional<get_configs_response>>
+        parser_result;
+    if (serialized_doc.Parse(body).HasParseError()) {
+        parser_result.first = remote_config_parser_result::invalid_json;
+        return parser_result;
+    }
+
+    rapidjson::Value::ConstMemberIterator target_files_itr;
+    rapidjson::Value::ConstMemberIterator client_configs_itr;
+    rapidjson::Value::ConstMemberIterator targets_itr;
+
+    // Lets validate the data and since we are there we get the iterators
+    parser_result.first = validate_field_is_present(serialized_doc,
+        "target_files", rapidjson::kArrayType, target_files_itr,
+        remote_config_parser_result::target_files_field_missing,
+        remote_config_parser_result::target_files_field_invalid_type);
+    if (parser_result.first != remote_config_parser_result::success) {
+        return parser_result;
+    }
+    parser_result.first = validate_field_is_present(serialized_doc,
+        "client_configs", rapidjson::kArrayType, client_configs_itr,
+        remote_config_parser_result::client_config_field_missing,
+        remote_config_parser_result::client_config_field_invalid_type);
+    if (parser_result.first != remote_config_parser_result::success) {
+        return parser_result;
+    }
+    parser_result.first = validate_field_is_present(serialized_doc, "targets",
+        rapidjson::kStringType, targets_itr,
+        remote_config_parser_result::targets_field_missing,
+        remote_config_parser_result::targets_field_invalid_type);
+    if (parser_result.first != remote_config_parser_result::success) {
+        return parser_result;
+    }
+
+    auto [result_parse_tf, target_files] = parse_target_files(target_files_itr);
+    parser_result.first = result_parse_tf;
+    if (parser_result.first != remote_config_parser_result::success) {
+        return parser_result;
+    }
+
+    auto [result_parse_cc, client_configs] =
+        parse_client_configs(client_configs_itr);
+    parser_result.first = result_parse_cc;
+    if (parser_result.first != remote_config_parser_result::success) {
+        return parser_result;
+    }
+
+    auto [result_parse_targets, targets] = parse_targets(targets_itr);
+    parser_result.first = result_parse_targets;
+    if (parser_result.first != remote_config_parser_result::success) {
+        return parser_result;
+    }
+
+    parser_result.second = get_configs_response(
+        std::move(client_configs), target_files, std::move(targets.value()));
+
+    return parser_result;
+}
+// NOLINTNEXTLINE(cppcoreguidelines-macro-usage)
+#define RESULT_AS_STR(entry) #entry,
+constexpr static std::array<std::string_view,
+    (size_t)remote_config_parser_result::num_of_values>
+    results_as_str = {PARSER_RESULTS(RESULT_AS_STR)};
+std::string_view remote_config_parser_result_to_str(
```

Comment:
This is unused within the remote config code, do we need it?

Reply:
This functions is not used in production code. However it becomes handy when doing tests and getting failures